kong/db/iteration.lua

```diff
@@ -15,7 +15,7 @@ end
 
 local function page_iterator(pager, size, options)
   local page = 1
-  size = size or 100
+  size = size or 1000
```

Comment:
It is a bit unfortunate that the C* strategy uses this low-level module for cascade deletes, because it prevents us from being able to specify the default fallback (`1000`) in the DAO module only, making it authoritative, and enabling a few other nice-to-haves.

Besides, the approach of having the constants module loaded all of our DAO/strategies/cluster events modules isn't ideal either, as it reduces encapsulation of all of these modules and prevents us from **configuring** this property (at _runtime_), which is still the ultimate goal we should be looking at here.

---

The structure I had in mind when looking at #4770 was to have the `kong.db` module be the authoritative source for default values such as `default_page_size` and `max_page_size`. E.g. a table stored as an attribute: `kong.db.defaults`, and given when instantiating it:

```lua
-- kong/init.lua
local db = DB.new(config, nil, {
  page_size = constants.default_page_size,
  max_page_size = constants.max_page_size,
})
```

or:

```lua
-- kong/db/init.lua
function DB.new(kong_config, strategy)
  -- ...

  local self = {
    -- ...
    defaults = {
      page_size = constants.default_page_size,
      max_page_size = constants.max_page_size,
    },
  }

  -- ...
end
```

Why?

- It allows us to give a reference to this table to the DAOs (`kong.db.dao`) and the strategies or low-level helpers (`kong.db.iteration`) as needed.
- In the future, the values in this table can be updated at runtime as necessary based on the size of the database tables.
- In `options` table of all of our DAO functions, we accept a new `options.defaults` table. Then the DAO's fallback code in `each()` and `each_for_*()` becomes: `page_size = options.defaults.page_size or self.defaults.page_size`. 
- The [Admin API DAO options](https://github.com/Kong/kong/blob/master/kong/api/endpoints.lua#L127-L129) could would contain a new `defaults = { page_size = 100, max_page_size = 1000 }` attribute, given to all of the `each()` calls in the Admin API.
- In this low-level `kong.db.iteration` module, we `assert(page_size ~= nil)`, and the Cassandra strategy's cascade delete code grabs the `self.defaults.page_size` value, which can never be `nil`.

Reply:
@thibaultcha I tried to do that, but there were some dependencies, that would take time to figure out just right, and I could not finish on that. In this current shape I think we at least get a bit closer to the goal.